vanilla-todo/src/core/Component.js

```diff
@@ -0,0 +1,62 @@
+import { frozen } from "../util/fronzen.js";
+import { observable, observe } from "./observer.js";
+
+class Component {
+  $root;
+  #state;
+  #props;
+  constructor($root, props) {
+    this.$root = $root;
+    this.#props = props;
+    this.#state = observable(this.initState());
+    observe(() => {
+      this.#render();
+      this.#setEvent();
+      this.mounted();
+    });
+  }
+  get state() {
+    return frozen({ ...this.#state });
+  }
+  get props() {
+    return frozen({ ...this.#props });
+  }
+  setState(newState) {
+    for (const [key, value] of Object.entries(newState)) {
+      if (!this.#state.hasOwnProperty(key)) continue;
+      this.#state[key] = value;
+    }
+  }
+  #render() {
+    this.$root.innerHTML = this.html() || "";
+  }
+  #addEvent(eventType, targetSelector, callback) {
+    const listener = (event) => {
+      if (!event.target.closest(targetSelector)) {
+        return;
+      }
+      callback(event);
+    };
+
+    if (this.$root) {
+      this.$root.addEventListener(eventType, listener);
+    }
+  }
+  #setEvent() {
+    const events = this.event();
+    if (!events) {
+      return;
+    }
```

Comment:
```suggestion
    const events = this.event() || [];
```
이러면 한 방에 해결되지 않을까요!?